### DISCUSSION TITLE
Synonym modeling

### DISCUSSION THREAD

**Discussion by axmng** [MaartenGr/BERTopic#1343] [Q&A] [answered]
Is it possible to use BERTopic to only cluster synonyms and leave terms with no direct semantical match as standalone topic?

My aim it to cluster texts on a very fine-granular level. 

I was looking into `min_topic_size=2`  (with HDBSCAN) but the topic span of each topic is still to broad for my use case. 
Not sure if it makes sense but I was wondering if something like `max_topic_size` would be desirable.?
Or can I define a minimum value for the similarity of terms within a cluster?

Many thanks!

**Comment by MaartenGr** [comment, marked as answer]
I am not sure whether HDBSCAN is the right solution for you. If you want to make sure they have a very strict relationship, then using something like [AgglomerativeClustering](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.AgglomerativeClustering.html#sklearn.cluster.AgglomerativeClustering) where you can define a certain threshold would be more appropriate.

**Reply by axmng** [reply]
Great! Many thanks. This helps.
